server/src/main/java/io/druid/segment/realtime/appenderator/FiniteAppenderatorDriver.java

```diff
@@ -91,6 +90,11 @@
   // Notified when segments are dropped.
   private final Object handoffMonitor = new Object();
 
+  // SegmentIdentifier -> Number of Summarized rows in the segment
+  // Used by Kafka Indexing Task to create new appenderator driver when limit is reached
+  private final Map<String, Integer> rowsInSegment = Maps.newHashMap();
```

Comment:
Also, I think it would be better to move this to DriverHolder because it is the only class which is using this information.